client/client-lib/src/api.rs

```diff
@@ -557,6 +559,8 @@ impl<C: JsonRpcClient> WsFederationApi<C> {
                                 futures.push(member.request(method, &params));
                             }
                         }
+                        sleep(Duration::from_millis(delay_ms)).await;
```

Comment:
We process here independent responses from all members, and this is a success case. Is this the right place to sleep? I would expect this should be done in the place that does the per-member retry (somewhere lower level). Or in the higher level (the whole call to the federation failed). Or I'm just confused.

Reply:
That's a good question, an error/success is actually only determined in the `QueryStep::Finished(result)` case based upon how the `QueryStrategy` handles it.

The `QueryStep::Request(peers)` is when the strategy wants to retry sending requests to peers, that's why I stuck the sleep here.

Every request is automatically sent to multiple peers which is why the `QueryStep::Continue` case exists, but we don't want to sleep between those since they aren't retries, just due to our trust model we always need to query multiple peers.

Reply:
> That's a good question, an error/success is actually only determined in the `QueryStep::Finished(result)` case based upon how the `QueryStrategy` handles it.


I'm not sure if `Request` is just confusingly named, or is `Request` only used to retry stuff on failure or does it ever happen for valid reasons? If so maybe it should be named `Retry`?

Reply:
Yes, we could rename it to `Retry`